test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -62,15 +62,18 @@
   ([env-name if-none-match]
    (let [opts (if if-none-match
                 {:headers {"If-None-Match" if-none-match}})]
-     (http-client/get
-      (str "https://localhost:8140/puppet/v3/"
-           "environment_classes?"
-           "environment="
-           env-name)
-      (merge
-       testutils/ssl-request-options
-       {:as :text}
-       opts)))))
+     (try
+       (http-client/get
+        (str "https://localhost:8140/puppet/v3/"
+             "environment_classes?"
+             "environment="
+             env-name)
+        (merge
+         testutils/ssl-request-options
+         {:as :text}
+         opts))
+       (catch Exception e
+         (throw (Exception. "environment_classes http get failed" e)))))))
```

Comment:
Wrapping the raw exception we get from the HTTP client with an exception that will have the stack trace that this line was called from.  HTTP client exceptions typically occur on Apache worker threads so you can't tell what line of code in the calling test was active on the stack otherwise.